common/math/integer.go

```diff
@@ -0,0 +1,25 @@
+package math
+
+import gmath "math"
+
+/*
+ * NOTE: The following methods need to be optimised using either bit checking or asm
+ */
+
+// IsMinSafe returns whether the subtraction is safe and does not overflow.
```

Comment:
Should be

// SafeSub returns subtraction result and whether overflow occurred